extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosExporter.java

```diff
@@ -342,21 +347,41 @@ private List<PhotoModel> convertPhotosList(
     }
 
     for (GoogleMediaItem mediaItem : mediaItems) {
-      if (mediaItem.getMediaMetadata().getPhoto() != null) {
-        // TODO: address videos
-        boolean shouldUpload = albumId.isPresent();
-
-        if (tempMediaData != null) {
-          shouldUpload = shouldUpload || !tempMediaData.isContainedPhotoId(mediaItem.getId());
-        }
+      // TODO: address videos
+      boolean shouldUpload = albumId.isPresent();
```

Comment:
should we do this on a per-GoogleMediaItem basis to keep the code simpler?

Maybe a 

private static Optional<PhotoModel> convertToPhotoModel(
  Monitor monitor,
  UUID jobId,
  TempMediaData tempMediaData,
  GoogleMediaItem mediaItem,
   ...) {
}

Reply:
Ok, I pulled out the shouldUpload & photo model conversion logic. 

I left the monitoring in the original method, because I think the caller of the method should be the one logging, rather than hard coding logging in the method. 

I also decided to not make it Optional, and rather throw a specific custom exception. Let me know if you dont like this, I can go the optional route.